### ISSUE TITLE
md-swipe not working correctly on chrome for android

### ISSUE THREAD

**Issue by wolfet** [angular/material#10145]
**Actual Behavior**:
 -  `What is the issue? *` 
On the latest version of Chrome for Android, md-swipe is not working as expected. md-swipe-left is triggered from a swipe in any direction. md-swipe-right, md-swipe-up, and md-swipe-down do not respond.
 -  `What is the expected behavior?` 
md-swipe-left should respond only to left swipes, and other md-swipe-* directives should respond to their respective directions.
**CodePen** (or steps to reproduce the issue): *
 -  `CodePen Demo which shows your issue:` http://codepen.io/twolfe/pen/pNQBrb
 -  `Details:` This works on chrome desktop, but not chrome for Android version 55.0.2883.91. Open the codepen, http://codepen.io/twolfe/full/pNQBrb/, on an Android phone with the latest Chrome update, or with Chrome DevTools device mode, to see the issue. This issue is showing up on my Pixel XL.

**Angular Versions**: *
 -  `Angular Version:` 1.5.5
 -  `Angular Material Version:` 1.1.1

**Additional Information**:
 -  `Browser Type: *` Chrome for Android
 -  `Browser Version: *`55.0.2883.91
 -  `OS: *`Android 7.1.1
 -  `Stack Traces:`

**Comment by RoniPa** [comment]
This can also be repeated in the Chrome DevTools Device Mode

**Comment by LohithAnand** [comment]
I encountered the same issue on my hybrid Cordova app. Downgrading the Android System Web view from the play store is a workaround for hybrid apps which use web views for now. 
I hope the issue will be resolved soon.

**Comment by mdtjn** [comment]
Hi Team, its driving my users crazy. Need some quick fix

**Comment by imromec** [comment]
@LohithAnand, how to owngrade the Android System Web view from the play store?

**Comment by LohithAnand** [comment]
@imromec Locate the Android System Webview in play store and click on the uninstall button. This will rollback to the version of Webview shipped with the OS.

**Comment by SlavaSlimmy** [comment]
Same problem on main site demo in the Chrome DevTools Device Mode: https://material.angularjs.org/latest/demo/swipe

**Comment by mohammadrafigh** [comment]
+1

**Comment by motabass** [comment]
+1
Any workarounds for now?

**Comment by mohammadrafigh** [comment]
let me mention that issue also appears on Firefox when testing in device mode with touch enabled. it's kind of a normal to high priority issue IMO but there is no attention for it yet...

**Comment by watat83** [comment]
I'm currently building a web app which makes use of sliders extensively. Since I discovered this issue I can't find sleep. Development has slowed down as a result of this feature not working correctly especially in mobile. It work fine on desktop though.

**Comment by imromec** [comment]
If you are building hybrid app using cordova, then this issue can be resolved by installing cordova crosswalk plugin.

**Comment by RoniPa** [comment]
Seems that in Chrome Device Mode, in the $mdGestures provider, the touch pointercancel event with properties x and y as 0  is triggered on swipe end. In regular mode the mouse pointerup event is triggered with the x and y properties set to the location where mouse click ended. I believe the issue could have something to do with this.

**Comment by duncan1a** [comment]
+1 ... really need a fix for this

**Comment by duncan1a** [comment]
Could the issue be here?

function getEventPoint(ev) {
  ev = ev.originalEvent || ev; // support jQuery events
  return (ev.touches && ev.touches[0]) ||
    (ev.changedTouches && ev.changedTouches[0]) ||
    ev;
}


ev.touches and ev.changedTouches are always undefined so ev is always returned. 

I'm not sure how ev.touches are supposed to work, but it seems that this is looking for the original event to return. That is compared in updatePointerState() with the current event and direction calculated.

Any ideas on how to get around this?

**Comment by RoniPa** [comment]
I believe the issue might be in the missing pageX and pageY properties on the last returned touch event (pointercancel), because the swipe direction gets always calculated towards the upper left corner here:

```
/*
 * Update the given pointer based upon the given DOMEvent.
 * Distance, velocity, direction, duration, etc
 */
function updatePointerState(ev, pointer) {
  var point = getEventPoint(ev);
  var x = pointer.x = point.pageX;
  var y = pointer.y = point.pageY;

  pointer.distanceX = x - pointer.startX;
  pointer.distanceY = y - pointer.startY;
  pointer.distance = Math.sqrt(
    pointer.distanceX * pointer.distanceX + pointer.distanceY * pointer.distanceY
  );

  pointer.directionX = pointer.distanceX > 0 ? 'right' : pointer.distanceX < 0 ? 'left' : '';
  pointer.directionY = pointer.distanceY > 0 ? 'down' : pointer.distanceY < 0 ? 'up' : '';

  pointer.duration = +Date.now() - pointer.startTime;
  pointer.velocityX = pointer.distanceX / pointer.duration;
  pointer.velocityY = pointer.distanceY / pointer.duration;
}
```

As this broke with a Chrome update I believe they changed the browser so that either the pointercancel event data changed or it's the wrong event type to begin with (are we expecting pointerup for example?)

**Comment by duncan1a** [comment]
RoniPa, yep, I think you're right. There is pointer.startX, what we need it the finishing point.

This is also messing up scrolling in my sidenav. It triggers the nav to close.

**Comment by ginalx** [comment]
This is a tricky one indeed. The very last **event** brings no mouse position info and the direction ends up always 'left'. Other touch options are getting a conflict, ng-swipe doesn't seem to trigger, ng-mouseup or down not working for touch move(for a fast manual solution).
Isn't this a Chrome error, messing up this event?

Just wanted to add a +1 here.

**Comment by mtf2000** [comment]
+1. This breaks the md-swipe-content in tabs as well. Also the demo is broken and only swipe left works.

**Comment by vjku** [comment]
+1 Looking forward to see this fixed soon.

If your development slowed down with this bug like ours, a quick workaround would be using https://github.com/adzialocha/angular-swipe and changing md-swipe-* to ng-swipe-* until this is fixed.

**Comment by ginalx** [comment]
Confirmed that angular-swipe does work.

**Comment by imjoeco** [comment]
Just ran into this after upgrading to Chrome Version 55.0.2883.87 on Ubuntu 16.04. It appears to be an issue with the chrome browser, not Angular Material. Swipe gestures still function fine in Firefox. I've submitted an error report to the Chrome team.

**Comment by mohammadrafigh** [comment]
@imjoeco maybe this is related to something in their js engine, but if the problem is with chrome or webview then why other libs out there detecting gestures correctly? ok maybe an api issue... but you can see that it's not working in Firefox in device (responsive) mode too.

**Comment by duncan1a** [comment]
@mohammadrafigh I think it's because material angular is using the 'pointercancel' event to detect the last position. A simple fix may just be to use another event. I want to have a look through some other swipe libs to see what they use, when I have some time

**Comment by sudhakar183** [comment]
+1 I am facing this issue with my Cordova application. Looking forward to see some fix soon.

**Comment by imjoeco** [comment]
@mohammadrafigh, I'm definitely not saying there isn't at least one work-around, I understand other libraries still work. My thinking is just that, unless angular material is using some deprecated browser api, it's essentially a bug with chrome, so in the effort of a standards compliant web and not doing more work than I need to, my starting point is to report the chrome bug to the chromium team, despite potential workarounds for the problem. To be clear, I'm not sure exactly the source of the issue, but as it predictably happened on browser update, I'm going to make sure the simplest explanation isn't the answer before trying to fix what not be broken.

**Comment by RoniPa** [comment]
I think I found a workaround for this. The issue seems to be that Chrome launches a pointercancel event when ever some browser default action is triggered. You can control this with the [touch-action](https://developer.mozilla.org/en-US/docs/Web/CSS/touch-action) CSS property. By disabling the conflicting default actions (for example, if you only need horizontal scrolling and have a md-swipe-right set the property as `touch-action:pan-y` ) I was able to prevent the unwanted pointercancel events and get the md-swipe actions working correctly. This doesn't however solve the issue entirely, but it seems to be more of a problem of how Chrome 55 has implemented the new PointerEvent API. [This article gave me some insight.](https://developers.google.com/web/updates/2016/10/pointer-events)

If the library is wished to be compatible with this, some changes might be needed to the way Angular Material groups events. I was also able to get this to work by ignoring the PointerEvents and reading only the old Mouse and Touch APIs, but that's probably not a very sustainable solution in the long run.

**Comment by RoniPa** [comment]
Except it seems that unwanted swipe-left actions are still triggered, so pointercancel events should probably be ignored in the gestureEnd function as to my understanding the intended purpose for this event is to roll back the gesture without triggering actions?

**Comment by iamroof** [comment]
Hi Team, This issue is causing problems in other swipe functionalities too. I have a map in my hybrid mobile application. In which, Zoom in/Zoom out is not working. Need some quick fix. Thanks

**Comment by mustaqahmed** [comment]
@clshortfuse and @RoniPa: This is not a bug in Chrome, looks like Angular needs a little fix here for PointerEvents. Simply add a style <code>touch-action: none</code> for the elements that will be swiped. I confirmed that it fixes the problem in Chrome.

Edge also acts strangely on Surface with the demo, and works perfectly with the suggested fix.

Here is an explanation for this behavior: We are seeing a <code>pointercancel</code> event here because the browser is taking over the control to handle the gestures here, as specified in the [PointerEvent spec](https://w3c.github.io/pointerevents/#the-pointercancel-event). To suppress the browser action (and let Angular handle it), we need to specify using <code>touch-action</code> which browser-actions are allowed. In my solution above, I specified that browser shouldn't do _anything_ for gestures starting on the swiped elements. This could be more granular than this, e.g. <code>touch-action: pan-x</code> would let browser do only horizontal panning/scrolling.

**Comment by clshortfuse** [comment]
@mustaqahmed To clarify, are you saying there isn't a bug with Chrome or this isn't a Chrome-exclusive bug?

**Comment by mustaqahmed** [comment]
Exactly: Chrome 55 shipped PointerEvent support, which exposed a hidden bug.

Angular seems to have a few other bugs around this. E.g. md-slider is working neither on [Android](https://github.com/angular/material/issues/10187) nor on [Windows touch](https://github.com/angular/material/issues/10294). This is true for Edge too.

The same fix works in both cases. I will comment on those issues.

**Comment by waldandrade** [comment]
Good Night, 

Why swipe-right works fine in Material Angular page, but not works in my application? I think there is somethink is broken it.

**Comment by waldandrade** [comment]
In my browser, if i inspect de webpage like a mobile, it dont work too.

**Comment by waldandrade** [comment]
Someone know if  is ngTouch broken too?

**Comment by RoniPa** [comment]
@mustaqahmed Yep, I think you're right. I created a fork of my own where I handle pointercancel and touchcancel separately from other 'ending' events. When combined with touch-action CSS property it seems to remove issues at least with Chrome. I haven't had time to test it out extensively though.

**Comment by waldandrade** [comment]
I'm testing angular-touch directive right now.

**Comment by waldandrade** [comment]
@RoniPa https://developers.google.com/web/updates/2016/10/pointer-events

**Comment by waldandrade** [comment]
Pointer Events

Pointing at things on the web used to be simple. You had a mouse, you moved it around, sometimes you pushed buttons, and that was it. But this, doesn’t work so well on here.

Touch events are good, but to support touch and mouse, you had to support two event models:

elem.addEventListener('mousemove', mouseMoveEvent);
elem.addEventListener('touchmove', touchMoveEvent);
Chrome now enables unified input handling by dispatching PointerEvents:

elem.addEventListener('pointermove', pointerMoveEvent);
Pointer events unify the pointer input model for the browser, bringing touch, pens, and mice together into a single set of events. They’re supported in IE11, Edge, Chrome, Opera and partially supported in Firefox.

Check out Pointing the Way Forward for more details.

**Comment by waldandrade** [comment]
The bad new is that angular material handles "pointermove". So i go back to 0 stage!

**Comment by waldandrade** [comment]
Above, other bad new on angular-material.js:

```
/**
 * Detect if the ng-Touch module is also being used.
 * Warn if detected.
 * @ngInject
 */
function DetectNgTouch($log, $injector) {
  if ( $injector.has('$swipe') ) {
    var msg = "" +
      "You are using the ngTouch module. \n" +
      "Angular Material already has mobile click, tap, and swipe support... \n" +
      "ngTouch is not supported with Angular Material!";
    $log.warn(msg);
  }
}
```

**Comment by waldandrade** [comment]
**Test 1: Is angular-material handling Chrome pointermove?**
 answer:
Yes

I did a simple test with console.log, and it works fine!

**Comment by waldandrade** [comment]
**Test 2: How angular material treats swipe?**
answer:
That's not a fail but a math error. On mobile or chrome inspection swipe left and swipe right are both recognized by 'left', when i close inspection the lib treats it right.

**Comment by waldandrade** [comment]
That's the swipe treatment:

```
  var point = getEventPoint(ev);
  var x = pointer.x = point.pageX;
  var y = pointer.y = point.pageY;

  pointer.distanceX = x - pointer.startX;
  pointer.distanceY = y - pointer.startY;
  pointer.distance = Math.sqrt(
    pointer.distanceX * pointer.distanceX + pointer.distanceY * pointer.distanceY
  );

  pointer.directionX = pointer.distanceX > 0 ? 'right' : pointer.distanceX < 0 ? 'left' : '';
```

Can you guys help me with it?

**Comment by waldandrade** [comment]
Conclusion 1:

```
m.Event {
 originalEvent: PointerEvent, 
  type: "pointercancel", timeStamp: 80260, 
jQuery111107948628520027305: true, 
toElement: md-content.md-padding.layout-wrap.ng-scope._md.layout-align-start-start.layout-row.flex-auto…}altKey: falsebubbles: truebutton: 0buttons: 0cancelable: falseclientX: 0clientY: 0ctrlKey: falsecurrentTarget: documentdata: undefineddelegateTarget: documenteventPhase: 3fromElement: nullhandleObj: ObjectisDefaultPrevented: bb()jQuery111107948628520027305: truemetaKey: falseoffsetX: 0offsetY: -64originalEvent: PointerEventpageX: 0pageY: 0relatedTarget: nullscreenX: 0screenY: 0shiftKey: falsetarget: md-content.md-padding.layout-wrap.ng-scope._md.layout-align-start-start.layout-row.flex-autotimeStamp: 80260toElement: md-content.md-padding.layout-wrap.ng-scope._md.layout-align-start-start.layout-row.flex-autotype: "pointercancel"view: nullwhich: 1__proto__: Object

distanceX: -344

directionX: left
```

There is a event 'pointercancel' like @RoniPa said.

**Comment by waldandrade** [comment]
There are two way:

pointerup pointercancel

In our conditions, chrome always call 'pointercancel', I suspect that distanceX negative can be a cause.

**Comment by waldandrade** [comment]
"With pointer events, whenever a default action like scroll or zoom is triggered, you’ll get a pointercancel event, to let you know that the browser has taken control of the pointer."

https://developers.google.com/web/updates/2016/10/pointer-events

**Comment by waldandrade** [comment]
Guys, try to move swipe to out of md-content elements.

**Comment by waldandrade** [comment]
The solution is more simple that it seems:

Browser take controle of elements when they want, but you can say what element you don't want give to browser control.
```
div{ 
    touch-action : pan-y;
}
```
It worked for me.

**Comment by mohammadrafigh** [comment]
@waldandrade thanks for your tips. I'm sorry but please update your conments instead of commenting multiple times this will spam others. Again thanks for your efforts.

**Comment by waldandrade** [comment]
@mohammadrafigh  it was my evolution step by step, i'm sorry about spam.

**Comment by uncledlm** [comment]
I have something that is similar but can't fix.  I have a view with 2 tabs.  When I scroll the left area it activates the tab to the right.  Structure is:

<mb-tabs> <mb-tab> <md-tab-body> <md-content> <md-list> <md-list-item> ...

I tried inserting the touch-action property but the tab keeps switching on the slightest scroll.

This only happens on Chrome for Android -- works fine on Chrome for IOS

**Comment by DevVersion** [comment]
@EladBezalel Is this solved now?

**Comment by mohammadrafigh** [comment]
@DevVersion swipe examples on website shows that Swipe is fixed generally but something is wrong still with tabs, md-swipe-content still has the issue.

**Comment by darkurse** [comment]
Same result as @mohammadrafigh. 
But strangely, if md-tabs is contained within a md-bottom-sheet, then swipe works fine.

**Comment by mtf2000** [comment]
I updated do 13 with high hopes but tabs are still sliding to the right.

**Comment by iamroof** [comment]
With pointer events, whenever zoom is triggered, It seems pointercancel event is triggered always. I have a map in my app which is not working(zoom in / zoom out) after chrome 55 update.

**Comment by waldandrade** [comment]
It is because browsers need to crontrol scroll. Try to turn off browser
control. I think there are more complex ways but that is the best way i
found.

Em 02/02/2017 09:40, "iamroof" <notifications@github.com> escreveu:

With pointer events, whenever zoom is triggered, It seems pointercancel
event is triggered always. I have a map in my app which is not working(zoom
in / zoom out) after chrome 55 update.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/angular/material/issues/10145#issuecomment-276946933>,
or mute the thread

**Comment by watat83** [comment]
Can this issue be solved? This is an important feature in Material Design and they don't even bother fixing it, when hundreds of not thousands of people are relying on the framework to run their business? This is ridiculous.

**Comment by Paul-Reed** [comment]
Come on guy's, this is a big problem for lots of users.
Visitors are currently unable to scroll down pages, and cannot access content.

**Comment by iamroof** [comment]
@waldandrade this isnt working. Tried turning off browser scroll control.

**Comment by trevorhreed** [comment]
The ngTouch module, built in to Angular doesn't have this problem. I added that module to my application and replaced `md-swipe-left` with `ng-swipe-left`. All is well again.

**Comment by SlavaSlimmy** [comment]
Try to scroll

**Comment by mohammadrafigh** [comment]
ok now I see scroll is broken inside pages with tabs too, new damn issue, it's funny that the latest commit in master is for 28 days ago

**Comment by mohammadrafigh** [comment]
this PR should fix this issue completely (issues with swipe, tabs and scrolling). I hope this works for everyone.

**Comment by EladBezalel** [comment]
@mohammadrafigh i'm looking into it

**Comment by theLufenk** [comment]
@mohammadrafigh @EladBezalel 

The PR is definitely in the right direction. However, the issue isn't really fixed.

Excluding the pointercancel event in the check for 'end-event' will not work due to the following reason.

**FROM MDN**
"The pointercancel event is fired when the browser concludes the pointer will no longer be able to generate events.  For example, when a touchscreen action was triggered that was not disallowed by the touch-action CSS property."

Chrome fires both pointerdown and touchstart initially. **However, it is not guaranteed that the "pointerup" event will always be fired at the end**. Infact, the pointercancel event maybe fired, resulting in the touch-move cycle being resetted (I dont know how to put this in better words, but my point is that the next tap action will be a pointerup/touchstart).This might happen due to a slow swipe or other reasons which I cannot figure-out. But just disabling the 'touch-action' css value does not guarantee that pointercancel will be supressed.

But interestingly, a touch-end is always fired at the end, causing me to believe that the POINTER API in itself is not reliable to detect precision swipes based on pointerdown-pointerup cycles, as a pointercancel can be fired at anytime.

I have tested extensively on chrome, both in Android as well as Responsive Mode on a Mac.
**A pointercancel event can be fired anytime**.

The whole issue is due to **'pointercancel'** event. Specifically due to this function:
`function typesMatch(ev, pointer) {
  return ev && pointer && ev.type.charAt(0) === pointer.type;
}`
The tab-change on scroll down occurs due to a pointerdown-pointercancel cycle returning true. Same for right swipe going to the next tab. **Hence, a pointerdown-pointercancel cycle is treated like a pointerdown-pointerup cycle, which is the root cause of the issue.**

**THE SOLUTION THAT WORKS FOR ME**
Drop listening to pointer events. For all practical purposes, the mouse and touch events suffice. Safari has no support for it.
The browser support for pointer events is almost not very amazing.
![screen shot 2017-03-06 at 1 10 20 pm 1](https://cloud.githubusercontent.com/assets/13207546/23601375/2d00a440-0271-11e7-9c94-25be8b5cd5ec.png)


**So all those who are struggling with tabs switching on scroll, or the swipe being in the right direction only, here's what you can do:**

Change the following:
// Listen to all events to cover all platforms.
  var START_EVENTS = 'mousedown touchstart pointerdown';
  var MOVE_EVENTS = 'mousemove touchmove pointermove';
  var END_EVENTS = 'mouseup mouseleave touchend touchcancel pointerup pointercancel';

To:
// Listen to all events to cover majority of platforms.
  var START_EVENTS = 'mousedown touchstar';
  var MOVE_EVENTS = 'mousemove touchmoev';
  var END_EVENTS = 'mouseup mouseleave touchend touchcancel';


Ironically, the purpose of pointer events is to provide a unified api for the mouse/touch events.However, we are listening to all these events, along with pointer events. **Further, the handler are attached to the Document element !**

Also, as per MDN:
**Touch-action property may be applied to all elements except: non-replaced inline elements, table rows, row groups, table columns, and column groups**
So, we cannot prevent panning,zooming,etc in these. So the solution with touch-action:pan-x/y is not all-inclusive.

**TL;DR**
Stop listening to pointer-events and your app will work just fine.

I request to kindly let me know in what Browser/Platform, will the touch/mouse events not be sufficient, as I am not listening to them in my production app. So it would be great to know where have I broken swipe gesture by doing this.

Thanks.

**Comment by RoniPa** [comment]
@theLufenk A while back when this problem first appeared after Chrome enabled the new pointer api I [created a fork](https://github.com/angular/material/compare/master...RoniPa:master) because of this issue where i separate the handling of cancel events. Does this kinda work towards a solution without needing to ignore the entire pointer api, which doesn't seem like a lasting fix?

**Comment by theLufenk** [comment]
@RoniPa 
Completely agree with you. This is in no way a lasting fix. However, I am pretty sure about the amount of headache it must be giving to the folks. So I have offered a temporary fix. I am unable to find out a input type (pen/stylus ?) which doesn't result in a touch/mouse event being fired. I request you to kindly let me know of I device which can be used to do a swipe action (from an abstract user's perspective, not technically ) and yet not produce a touch/mouse event. I am really interested in solving this problem but simply **couldn't see a scenario where my fix will fail**.


**Also, going through the w3c spec reveals:**
Examples of scenarios in which a user agent might determine that a pointer is unlikely to continue to produce events include:

1) A device's screen orientation is changed while a pointer is active.
2) The user inputs a greater number of simultaneous pointers than is supported by the device.
3) The user agent interprets the input as accidental (for example, the hardware supports palm rejection).
4) The user agent interprets the input as a pan or zoom gesture.

I would like to draw your attention to points number 2 and 3. They signify that the **pointer events get further dependant on the device hardware**

**Comment by theLufenk** [comment]
A important point to consider is that the pointer events API's main purpose was to provide a unified interface for accessing pointer interactions. Which directly relates to (as mentioned in the w3c draft)  **removing the need for developers to attach separate listeners for touch/mouse events, which end up affecting performance**

So one of the reason why pointerevents api was introduced is to reduce the cycles consumed by touch and mouse events used in same script. **This is an important takeway for us because:**  

1) I am not sure how accurate I am, but unless the dispatch of touch/mouse/pointer events is done, the browser does not take it's default action for zoom/pan etc. In essence, **by introducing another event listener for pointer events, we are further aggravating the issue which the pointer events api seeks to resolve.**

2) Event listeners do affect both performance and memory. And here, the listeners are not the document. Please add a console log inside the three functions which handle start,move and end in gesture.js. In non-mobile environments , the pointerevents listeners are essentially doubling the overhead, since they are never exclusive and fired along with either mouse or touch events.

Hence, IMHO using pointer events would only make sense when we can use **on('pointerdown')**  in place of **on('touchstart mousemove pointerdown')** as there is no apparent reasons or advantages of use pointerevents. There purpose is to enable us to listen to only 1 event. And ironically, we are using 3. So instead of reducing from 2 to 1, we are going from 2 to 3.

**I hope my point is clear enough**

**To the maintainers:** Please see it makes sense to remove the pointerevents, so that the users can still serve the framework from CDNs instead of having to include it locally just for a change encompassing 5 words. Also, if you guys can point out why they were included in the first place, it would be great !

Thank you

**Comment by RoniPa** [comment]
@theLufenk I can see your point and agree that for now that might actually be the solution that makes the most sense, as touch and mouse events aren't probably going to be deprecated any time soon in major browsers.

**Comment by theLufenk** [comment]
@RoniPa 
Sadly, IE11+ do not use touchevents but use pointerApi !!

**Comment by EladBezalel** [comment]
So i'm checking against the latest version and it seems to be working :
http://codepen.io/EladBezalel/pen/ZeodPy

**Comment by mohammadrafigh** [comment]
@EladBezalel md-swipe is working but as you see it's not working with tabs specially when the content of tabs has scrolls.

**Comment by iagocaldeira** [comment]
My tabs scroll still not working

**Comment by stephengardner** [comment]
This issue is kind of funny.

Lot of annoyed people.  

I actually didn't notice this at first and it caused our entire user profile system to be unscrollable on mobile.

Yikes.

**Comment by EladBezalel** [comment]
will be solved by @mohammadrafigh in #10455

**Comment by sdpillay** [comment]
Any update when it will be resolved?

**Comment by SensationSama** [comment]
Using ngTouch instead seems to fire events as desired.

https://docs.angularjs.org/api/ngTouch

**Comment by kylemacfarlane** [comment]
The fix in 8c1ce1b breaks pinch zoom due to the `touch-action: pan-x/y` applied to elements.

It's particularly bad with `md-tabs` as even if you don't have `md-swipe-content` set to true it will still have `md-swipe-left/right` attributes.

Annoyingly, pinch zoom works with shift+click in Chrome dev tools touch emulation so you have to test on an actual Android device to recreate the bug.

**Comment by Splaktar** [comment]
Fixed in https://github.com/angular/material/pull/10455 which was in `1.1.5`.
